lib/mathjax-helper.coffee

```diff
@@ -43,15 +44,72 @@ module.exports =
     return
 
 #
+# Define some functions to help get a hold of the user's Latex
+# Macros.
+#
+namePattern = ///             # The name of a macro can be either
+              ^[^a-zA-Z\d\s]$ # a single non-alphanumeric character
+              |               # or
+              ^[a-zA-Z]*$     # any number of lower and upper case
+              ///             # letters, but no numerals.
+
+valuePattern = ///^    # The stringified version of the definition
+               {.*}    # must be between brackets
+               (,\d)?  # and may be followed by a comma and a digit
+               $///
+
+getUserMacrosPath = ->
+  userMacrosPath =  CSON.resolve(path.join(atom.getConfigDirPath(),'markdown-preview-plus'))
+  userMacrosPath ? path.join(atom.getConfigDirPath(),'markdown-preview-plus.cson')
+
+loadMacrosFile = (filePath) ->
+  return {} unless CSON.isObjectPath(filePath)
+  CSON.readFileSync filePath, (error, object={}) =>
+    if error?
+      console.warn "Error reading Latex Macros file '#{filePath}': #{error.stack ? error}"
+      atom.notifications?.addError("Failed to load Latex Macros from '#{filePath}'", {detail: error.message, dismissable: true})
+    object
+
+loadUserMacros = ->
+  userMacrosPath = getUserMacrosPath()
+  if fs.isFileSync(userMacrosPath)
```

Comment:
I ended up keeping this `isFileSync` because it's a convenient place to include generation of the template file. I tried putting it in the `CSON.isObjectPath` but thought a file check was more general and readable . . . agree?